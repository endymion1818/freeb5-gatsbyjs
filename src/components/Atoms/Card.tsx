import React, { FC } from "react";
import styled, { css } from "styled-components";

import * as token from "../tokens";

export interface ICardProps {
  /**
   * text alignment
   * @default token.ETEXTALIGN.LEFT
   */
  textAlign?: token.ETEXTALIGN;
  /**
   * children of this component
   * @default <>test</>
   */
  children: string | JSX.Element[] | JSX.Element;
  /**
   * background colour
   * @default token.EBACKGROUND_COLOUR.SURFACE
   */
  backgroundColour?: token.EBACKGROUND_COLOUR;
  /**
   * text colour
   * @default token.ETEXT_COLOUR.ON_SURFACE
   */
  textColour?: token.ETEXT_COLOUR;
  /**
   * text colour
   * @default token.ETEXT_COLOUR.ON_SURFACE
   */
  linkColour?: token.ETEXT_COLOUR;
}

export const CardCss = css<ICardProps>`
    padding: ${token.ESIZE.SINGLE};
    display: flex;
    flex-direction: column;
    justify-content: space-between;
    margin-bottom: ${token.ESIZE.SINGLE};
    ${({ textAlign }) => textAlign && `text-align: ${textAlign};`}
    border-radius: ${token.EBORDERRADIUS.MEDIUM};
    background-color: ${({ backgroundColour }) =>
      backgroundColour ? backgroundColour : `rgba(0, 0, 0, 0.5);`};
    color: ${({ textColour }) =>
      textColour ? textColour : token.ETEXT_COLOUR.ON_SURFACE};

    a:not(.button) {
        color: ${({ linkColour }) =>
          linkColour ? linkColour : token.ETEXT_COLOUR.ON_SURFACE};
    }

    @media (min-width: ${token.EBREAKPOINT.MEDIUM}) {
        min-height: 28rem;
    }
`;

const SCard = styled.div`
  ${CardCss}
`;

const Card: FC<ICardProps> = ({
  children,
  textAlign,
  backgroundColour,
  textColour,
  linkColour
}) => {
  if (!children) {
    return null;
  }
  return (
    <SCard
      textAlign={textAlign}
      backgroundColour={backgroundColour}
      textColour={textColour}
      linkColour={linkColour}
    >
      {children}
    </SCard>
  );
};

export default Card;
