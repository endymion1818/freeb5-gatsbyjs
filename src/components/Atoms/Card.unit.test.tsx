import { cleanup, render } from "@testing-library/react";
import React from "react";
import Card from "./Card";

const CardDefaultProps = {
  children: "test"
};

describe("<Card />", () => {
  beforeEach(cleanup);
  it("Should render", () => {
    const { container } = render(<Card {...CardDefaultProps} />);
    expect(container).toMatchSnapshot();
  });
});
