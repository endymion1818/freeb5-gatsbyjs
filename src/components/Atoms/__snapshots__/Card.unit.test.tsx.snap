// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> Should render 1`] = `
.c0 {
  padding: 1rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  margin-bottom: 1rem;
  border-radius: 0.5rem;
  background-color: rgba(0,0,0,0.5);
  color: white;
}

.c0 a:not(.button) {
  color: white;
}

@media (min-width:64em) {
  .c0 {
    min-height: 28rem;
  }
}

<div>
  <div
    class="c0"
  >
    test
  </div>
</div>
`;
